test-utils/runtime-tester/src/run_test.rs

```diff
@@ -86,6 +90,26 @@ pub struct Scenario {
     pub network_config: NetworkConfig,
     pub blocks: Vec<BlockConfig>,
     pub use_in_memory_store: bool,
+
+    /// Path to the on-disk home directory; relevant only if
+    /// `use_in_memory_store` is `false`.
+    ///
+    /// If this is `None` (and scenario is configured with an on-disk store) the
+    /// scenario will be run with a temporary directory as home directory.  Once
+    /// the scenario finishes, that temporary directory is deleted.  On the
+    /// other hand, if this is set it is caller’s responsible to clean the
+    /// directory *before and after* the test runs.
+    ///
+    /// This is useful when wanting to save the storage on-disk and investigate
+    /// it after the test runs and it’s also why the directory is not cleared.
+    ///
+    /// Note that this field is not serialised thus a scenario read from file
+    /// won’t have this path set.  This is motivated partially by security
+    /// considerations since if this could be stored in JSON file than user
+    /// might be tricked to run a test against `/home/foo/.near/mainnet`
+    /// directory.
+    #[serde(skip_serializing)]
+    pub home_dir: Option<std::path::PathBuf>,
```

Comment:
`#[serde(skip_serializing)]`  feels like a smell. What if we had two `run` methods instead: `run(self)`, `run_with_home_dir(self, home: &Path)`?.

Reply:
I don’t really like that ’cuz then we end up with combinatoric explosion of different `run` methods. How about just always returning the temporary directory and letting the caller decide whether they want to use it or ignore.

Reply:
> How about just always returning the temporary directory and letting the caller decide whether they want to use it or ignore.

Genius idea!